wolfcrypt/src/evp.c

```diff
@@ -6878,6 +6878,446 @@ int wolfSSL_EVP_get_hashinfo(const WOLFSSL_EVP_MD* evp,
+
+        for ( i = 0; cpysz > 0 && inLen > 0;i++) {
+            if ((res = Base64_SkipNewline(in, &inLen, &j))
+                                            == ASN_INPUT_E) {
+                return -1;  /* detected an illegal char in input */
+            }
+            e0 = in[j++];
+
+            if (e0 == '=')
+                pad = 1;
+
+            *(ctx->data + ctx->remaining + i) = e0;
+            inLen--;
+            cpysz--;
+        }
+
+        outsz = sizeof(ctx->data);
+        res = Base64_Decode( ctx->data, BASE64_DECODE_BLOCK_SIZE, out, &outsz);
+        if (res == 0) {
+            *outl += outsz;
+            out   += outsz;
+
+            ctx->remaining = 0;
+            XMEMSET(ctx->data, 0, sizeof(ctx->data));
+        }
+        else {
+            *outl = 0;
+            return -1;   /* return with error */
+        }
+    }
+
+    /* process data in input buffer */
+    while (inLen > 3) {
+        int pad3 = 0;
+        int pad4 = 0;
+
+        byte e[4];
+
+        if ((res = Base64_SkipNewline(in, &inLen, &j)) != 0) {
+            if (res == BUFFER_E) {
+                break;
+            }
+            else {
+                *outl = 0;
+                return -1;
+            }
+        }
+        e[0] = in[j++];
+        if (e[0] == '\0') {
+            break;
+        }
+        inLen--;
+        if ((res = Base64_SkipNewline(in, &inLen, &j)) != 0) {
+            if (res == BUFFER_E) {
+                break;
+            }
+            else {
+                *outl = 0;
+                return -1;
+            }
+        }
+        e[1] = in[j++];
+        inLen--;
+        if ((res = Base64_SkipNewline(in, &inLen, &j)) != 0) {
+            if (res == BUFFER_E) {
+                break;
+            }
+            else {
+                *outl = 0;
+                return -1;
+            }
+        }
+        e[2] = in[j++];
+        inLen--;
+        if ((res = Base64_SkipNewline(in, &inLen, &j)) != 0) {
+            if (res == BUFFER_E) {
+                break;
+            }
+            else {
+                *outl = 0;
+                return -1;
+            }
+        }
+        e[3] = in[j++];
+        inLen--;
+
+        if (e[0] == '=')
+            pad = 1;
+        if (e[1] == '=')
+            pad = 1;
+        if (e[2] == '=') {
+            pad = 1;
+            pad3 = 1;
+        }
+        if (e[3] == '=') {
+            pad = 1;
+            pad4 = 1;
+        }
+        if (pad3 && !pad4) {
+            *outl = 0;
+            return -1;
+        }
+
+        /* decode four bytes */
+        outsz = sizeof(ctx->data);
+        res = Base64_Decode( e, BASE64_DECODE_BLOCK_SIZE, out, &outsz);
+        if (res == ASN_INPUT_E) {
+            *outl = 0;
+            return -1;
+        }
+
+        *outl += outsz;
+        out   += outsz;
+    }
+    /* copy left data to ctx */
+    if (inLen > 0) {
+        unsigned char el;
```

Comment:
Please use more descriptive variable name than "el", thanks.

Reply:
Removed variable "el" and used "c" instead.